src/StoreKeeper/WooCommerce/B2C/Tools/TaskHandler.php

```diff
@@ -599,4 +599,22 @@ private function reportFailedTask($task_id, $exception)
         // Create an error task in the task queue
         LoggerFactory::createErrorTask('task-in-queue-failed', $exception, $task_id, $metaData);
     }
+
+    /**
+     * Get all new tasks by storekeeper id.
+     */
+    public static function getTasksByStorekeeperId(int $storekeeperId): ?array
+    {
+        global $wpdb;
+
+        $select = TaskModel::getSelectHelper()
+            ->cols(['*'])
+            ->where('status = :status')
+            ->where('storekeeper_id = :storekeeper_id')
+            ->bindValue('status', self::STATUS_NEW)
+            ->bindValue('storekeeper_id', $storekeeperId)
+            ->orderBy(['date_created DESC']);
+
+        return $wpdb->get_results(TaskModel::prepareQuery($select), ARRAY_A);
```

Comment:
from the docs it looks like get_results may return object as well https://developer.wordpress.org/reference/classes/wpdb/get_results/#return

Reply:
Enforced result  to be array using `ARRAY_A` parameter.